src/System.Private.Reflection.Core/src/System/Reflection/Runtime/ParameterInfos/EcmaFormat/EcmaFormatMethodParameterInfo.cs

```diff
@@ -0,0 +1,116 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Reflection.Runtime.General;
+using System.Reflection.Runtime.CustomAttributes;
+
+using Internal.Reflection.Core;
+using Internal.Reflection.Core.Execution;
+
+using System.Reflection.Metadata;
+
+namespace System.Reflection.Runtime.ParameterInfos.EcmaFormat
+{
+    //
+    // This implements ParameterInfo objects owned by MethodBase objects that have an associated Parameter metadata entity.
+    //
+    internal sealed partial class EcmaFormatMethodParameterInfo : RuntimeMethodParameterInfo
+    {
+        private EcmaFormatMethodParameterInfo(MethodBase member, MethodDefinitionHandle methodHandle, int position, ParameterHandle parameterHandle, QSignatureTypeHandle qualifiedParameterTypeHandle, TypeContext typeContext)
+            : base(member, position, qualifiedParameterTypeHandle, typeContext)
+        {
+            _methodHandle = methodHandle;
+            _parameterHandle = parameterHandle;
+            _parameter = Reader.GetParameter(parameterHandle);
+        }
+
+        private MetadataReader Reader
+        {
+            get
+            {
+                Debug.Assert(QualifiedParameterTypeHandle.Reader is MetadataReader);
+                return (MetadataReader)QualifiedParameterTypeHandle.Reader;
+            }
+        }
+
+        public sealed override ParameterAttributes Attributes
+        {
+            get
+            {
+                return _parameter.Attributes;
+            }
+        }
+
+        public sealed override IEnumerable<CustomAttributeData> CustomAttributes
+        {
+            get
+            {
+                IEnumerable<CustomAttributeData> customAttributes = RuntimeCustomAttributeData.GetCustomAttributes(this.Reader, _parameter.GetCustomAttributes());
+                foreach (CustomAttributeData cad in customAttributes)
+                    yield return cad;
+
+                ParameterAttributes attributes = Attributes;
+                if (0 != (attributes & ParameterAttributes.In))
+                    yield return ReflectionCoreExecution.ExecutionDomain.GetCustomAttributeData(typeof(System.Runtime.InteropServices.InAttribute), null, null);
```

Comment:
Nit: I'd prefer you import the S.R.InteropServices namespace (or just those specific types if you don't want to haul in the entire namespace) rather than put fully qualified typenames here...